nix/pkgs/images/operators/default.nix

```diff
@@ -0,0 +1,50 @@
+# It would be cool to produce OCI images instead of docker images to
+# avoid dependency on docker tool chain. Though the maturity of OCI
+# builder in nixpkgs is questionable which is why we postpone this step.
+#
+# We limit max number of image layers to 42 because there is a bug in
+# containerd triggered when there are too many layers:
+# https://github.com/containerd/containerd/issues/4684
+
+{ stdenv
+, busybox
+, dockerTools
+, e2fsprogs
+, git
+, lib
+, xfsprogs
+, utillinux
+, mayastor-release
+, mayastor-debug
+}:
+let
+  versionDrv = import ../../../lib/version.nix { inherit lib stdenv git; };
+  version = builtins.readFile "${versionDrv}";
+  env = stdenv.lib.makeBinPath [ busybox xfsprogs e2fsprogs utillinux ];
+
+  # common props for all operator images
+  operatorImageProps = {
+    tag = version;
+    created = "now";
+    config = {
+      Env = [ "PATH=${env}" ];
+    };
+  };
+in
+{
+  # Release image of node operator.
+  node-image = dockerTools.buildLayeredImage (operatorImageProps // {
```

Comment:
I was under the impression from the PR description we were renaming this to `node`? Or is that only in `images.*`? Will we see a non-`image` variant of `node`?

Reply:
I was debating what to do here. I removed the `image` suffix from those that were prefixed by `images.*` only as they seemed redundant.

The reason I kept it for the others was that they weren't prefixed with `images.*` so I thought that I should be more explicit here. Really this is a result of me trying to reorganise the images directory. Perhaps I should just put them all into the same place and that would be clearer.

Reply:
I have removed the `-image` now, but I do make a distinction between `-service` and `-operator`.